pages/index.tsx

```diff
@@ -121,14 +130,30 @@ export default function Home() {
     }
   }, [loadedFeeds]);
 
+  const sortedArchive = useMemo(() => {
```

Comment:
This approach works.
But in order to make work as required 
> If 2 or more feeds are pinned, the last pinned goes to the top.

there should be some kind of index or priority in the list.

The easiest solution that comes to me (without updating all the feeds)
 is to give the newly pinned feed, a priority higher that the highest current.

So initially `feed.priority` is `undefined`.
When users pinned a feed, check in all feeds what is the highest priority.
`undefined` first time, and set `priority: highest ? highest + 1 : 1`

when the feed is unpinned, we can set priority to `undefined` or maybe `0`. Not sure.

And the sort is base on priority, it may happen that feeds have priority:

`[undefined, 1, 3, undefined]`

and sorted will be displayed as

`[3, 1, undefined, undefined]`